packages/cli/src/execute/execute.ts

```diff
@@ -73,7 +76,8 @@ export function parseAdaptors(plan: ExecutionPlan) {
     // But there are a couple of cases mostly in test, when validation is skipped,
     // when the array may not be set
     // It's mostly redundant nbut harmless to optionally chain here
-    job.adaptors?.forEach((adaptor) => {
+    const jobAdaptors = (job.adaptors || []).concat(resolvedAdaptors);
```

Comment:
If I have a workflow with two steps and different versions of adaptors - let's say `common@latest` and `common@1` - this will get us in trouble. Because versions of `common` will resolve to the latest.

A bit disappointing that I don't have a failing test for this :(

Reply:
Yup, that's true. Just tested and the trouble mentioned holds.

Reply:
Resolved!